### ISSUE TITLE
Inconsistent game state between python API and GUI

### ISSUE THREAD

**Issue by pjottos** [branpk/wafel#29]
I'm using the python API to play an m64 file and read game state at every frame, but i see different values in the wafel GUI than the API returns.

My code (slightly simplified):
```py
game = wafel.Game("libsm64/sm64_us.dll")
_, inputs = wafel.load_m64("120star.m64")
frames = []

for i, input in enumerate(inputs):
    game.advance()
    game.write("gControllerPads[0].button", input.buttons)
    game.write("gControllerPads[0].stick_x", input.stick_x)
    game.write("gControllerPads[0].stick_y", input.stick_y)

    obj_positions = []
    for j in range(0, 240):
        pos = [
            game.read(f"gObjectPool[{j}].oPosX"),
            game.read(f"gObjectPool[{j}].oPosY"),
            game.read(f"gObjectPool[{j}].oPosZ"),
        ]
        obj_positions.append(pos)

    frames.append({
        "random_seed": game.read("gRandomSeed16"),
        "obj_positions": obj_positions,
    })
```
120star.m64 is the 2012 120 star TAS.
After running the above i check the result:
```python
>>> print(frames[546]["obj_positions"][101])
[
    2.2172523,
    16.883919,
    -4414.45,
]
```
But the GUI shows this position:
```python
[
    2.4636788368,
    17.130346298,
    -4414.203613,
]
```

**Comment by branpk** [comment]
This:
```python
    game.advance()
    game.write("gControllerPads[0].button", input.buttons)
    game.write("gControllerPads[0].stick_x", input.stick_x)
    game.write("gControllerPads[0].stick_y", input.stick_y)
```

should be this:
```python
    game.write("gControllerPads[0].button", input.buttons)
    game.write("gControllerPads[0].stick_x", input.stick_x)
    game.write("gControllerPads[0].stick_y", input.stick_y)
    game.advance()
```

Could you try that and let me know if that fixes it?

**Comment by pjottos** [comment]
Yes, that turned out to be a part of the problem. To give some context, I'm making something like libsm64 that is thread safe and I need to verify the output. I had already tried to swap the order like that which caused the values to differ much more from the GUI values so I assumed that the way I had it was correct. 

Looking more closely at the output however, it seems that the `Game` instance has a global timer value of 1 right after being created. This makes sense because `frame` returns 0 and the same combination is observed in the GUI, but I had assumed the global timer would be 0 at this point. 

So to fix my issue i now read the game state before calling advance. I also updated the way I handle input in my project and that turned out to fix a desync i had been struggling with for the past 2 days.
